common/host_util.go

```diff
@@ -30,3 +32,19 @@ func GetLocalIp() string {
 	localIp, _ = gxnet.GetLocalIP()
 	return localIp
 }
+
+func HandleRegisterIPAndPort(url *URL) {
+	// if developer define registry port and ip, use it first.
+	if ipToRegistry := os.Getenv("DUBBO_IP_TO_REGISTRY"); len(ipToRegistry) > 0 {
+		url.Ip = ipToRegistry
+	}
+	if len(url.Ip) == 0 {
+		url.Ip = GetLocalIp()
+	}
+	if portToRegistry := os.Getenv("DUBBO_PORT_TO_REGISTRY"); len(portToRegistry) > 0 {
+		url.Port = portToRegistry
+	}
```

Comment:
Plus, I am thinking here needs a port inspection, for example, a port exceeded 65535.